### PR TITLE
Improve error handling.

### PR DESCRIPTION
**Pull request by danielballan**
Various improvements related to error handling, inspired by initial user
feedback. Includes unit tests. Tested interactively as well.

See commit messages for details.

### CODE REVIEW

File: databroker_pack/_pack.py [hunk 1]
```diff
@@ -281,21 +287,31 @@ def copy_external_files(target_directory, root, files):
...
-        new_files.append(shutil.copy2(filename, dest))
-    return new_root, new_files
+        try:
+            os.makedirs(dest.parent, exist_ok=True)
+            new_files.append(shutil.copy2(filename, dest))
+        except Exception:
+            logger.exception("Error while copying %r to %r", filename, dest)
+            if strict:
+                raise
+            failures.append(filename)
```
[the remark this answers is not in the document]
Comment (danielballan) [reply databroker_pack/_pack.py:313]: Aggregating tracebacks can get expensive. We log them three lines above.

File: databroker_pack/commandline/pack.py [hunk 2]
```diff
@@ -224,6 +225,18 @@ def main():
...
+        # We write the catalog file at the end because we need to collect
+        # information to write the root_map. We never overwrite existing files,
+        # so we will fail if catalog.yml exists already. In order to save time
+        # and fail early, check that the file exists.
+        catalog_file_path = pathlib.Path(args.directory, "catalog.yml")
+        if os.path.isfile(catalog_file_path):
+            print(
+                f"The file {catalog_file_path} exists. Specify an empty directory, or "
+                "a nonexistent one (which will be created by "
+                "databroker-pack)."
```
[the remark this answers is not in the document]
Comment (danielballan) [reply databroker_pack/commandline/pack.py:237]: I don't know how to do that. The `setup.py` is generally not included in a source installation.

[the remark this answers is not in the document]
Comment (danielballan) [reply databroker_pack/commandline/pack.py]: Actually, I found a way to parameterize this in 772bb92.

File: databroker_pack/_pack.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrakitin) [reply databroker_pack/_pack.py:313]: OK, thanks.

**Pull request merged**